include/nlohmann/json.hpp

```diff
@@ -3727,20 +3796,72 @@ class basic_json // NOLINT(cppcoreguidelines-special-member-functions,hicpp-spec
     with range checking
     @sa see @ref value() for access by value with a default value
 
-    @since version 1.0.0
+    @since version 1.0.0; template added in version 3.10.0
     */
-    const_reference operator[](const typename object_t::key_type& key) const
+    template < class KeyT, typename std::enable_if <
+                   !std::is_same<typename std::decay<KeyT>::type, json_pointer>::value&&
+#if defined(JSON_HAS_CPP_17)
+                   std::is_same<typename std::decay<KeyT>::type, std::string_view>::value&&
+#endif
+                   std::is_convertible<KeyT, typename object_t::key_type>::value, int >::type    = 0 >
+    const_reference operator[](KeyT && key) const
```

Comment:
It seems strange to me to support operator[] on const. std::map::operator[] is explicitly non-const.

Reply:
Agreed, I saw that myself when checking yesterday and it threw me off, since it's practically equivalent to `std::map::at`.

The reason why it's implemented for `std::string_view` is because it's already been part of the library for std::string, or whatever the JSON object key has been.

This just extends the existing API, afaict.

Reply:
The rationale is described here: https://json.nlohmann.me/features/element_access/unchecked_access/#notes